cmd/soroban-rpc/internal/db/event.go

```diff
@@ -0,0 +1,336 @@
+package db
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	sq "github.com/Masterminds/squirrel"
+
+	"github.com/stellar/go/ingest"
+	"github.com/stellar/go/support/db"
+	"github.com/stellar/go/support/log"
+	"github.com/stellar/go/xdr"
+)
+
+const (
+	eventTableName = "events"
+	firstLedger    = uint32(2)
+	MinTopicCount  = 1
+	MaxTopicCount  = 4
+)
+
+// EventWriter is used during ingestion of events from LCM to DB
+type EventWriter interface {
+	InsertEvents(lcm xdr.LedgerCloseMeta) error
+}
+
+// EventReader has all the public methods to fetch events from DB
+type EventReader interface {
+	GetEvents(
+		ctx context.Context,
+		cursorRange CursorRange,
+		contractIDs [][]byte,
+		topics [][][]byte,
```

Comment:
I still hate seeing this lmao, can you typedef it to like `NestedTopicArray` or something? :laughing: